pandas/io/parsers.py

```diff
@@ -2123,9 +2123,27 @@ def _clean_mapping(mapping):
         else:
             clean_dtypes = _clean_mapping(self.dtype)
 
-        return self._convert_to_ndarrays(data, self.na_values, self.na_fvalues,
-                                         self.verbose, clean_conv,
-                                         clean_dtypes)
+        # Apply NA values.
+        clean_na_values = {}
+        clean_na_fvalues = {}
+
+        if isinstance(self.na_values, dict):
```

Comment:
No, I perfectly understand.  I just haven't found a clean way to do it yet, and it seems like it would be best to do as a follow-up if that works.